### ISSUE TITLE
Move task responds with 204 NoContent even when overwrite is set to F

### ISSUE THREAD

**Issue by ChristophAtAdlos** [OpenMarshal/npm-WebDAV-Server#18]
On a move task configured with the F-Flag (do not overwrite), the webdav-server responds with a 204 Status. Should this not be a `201 Created`  status. identic to the copy task?

pursuant Copy.js (line 127) it does, but the real answer is 204:
```javascript
                                        else if (overridded)
                                            arg.setCode(WebDAVRequest_1.HTTPCodes.NoContent);
                                        else
                                            arg.setCode(WebDAVRequest_1.HTTPCodes.Created);
```

Could be this is related to the following issue.

EDIT:

Could this be in relation with this entries in resources?

```diff
     resources: 
      { '/': [Object],
        '/ONE': [Object],
        '/README.txt': [Object],
        '/README2.txt': [Object],
        '/README3.txt': [Object],
        '/moved': [Object],
        '/copies/README3.txt': [Object],
+        '/moved/README3.txt': [Object],
+        'C:\Git\MyProject\webdavroot\moved\README3.txt': undefined } },
```

Should `'/README3.txt'` not be removed from resources?
Should the entry `'/moved/README3.txt': [Object]` not be alone, without the him following absolute path?

Versions:
webdav-server: 2.3.3
node:  LTS 6.11.3
os. Windows 10 Pro x64

**Comment by ChristophAtAdlos** [comment]
This could be a possible fix, it's working for me. But it's not tested without the F flag.
```patch
Index: src/manager/v2/fileSystem/FileSystem.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/manager/v2/fileSystem/FileSystem.ts	(revision 0dccd8a91a5f7a2b4dac4dd996542a64e6a1fc1e)
+++ src/manager/v2/fileSystem/FileSystem.ts	(revision )
@@ -679,10 +679,14 @@
         const pathTo = new Path(_pathTo);
 
         const callback : ReturnCallback<boolean> = (e, overrided) => {
-            if(!e)
-                this.emit('move', ctx, pathFrom, { pathFrom, pathTo, overwrite, overrided })
-            callbackFinal(e, overrided);
-        }
+            const checkOverrideState : ReturnCallback<boolean> = function (exists?) {
+                overrided = (exists !== undefined) && overrided && overwrite; // Set the overrided state to a more real value
+                if(!e)
+                    this.emit('move', ctx, pathFrom, { pathFrom, pathTo, overwrite, overrided });
+                callbackFinal(e, overrided);
+            };
+            this.fastExistCheckExReverse(ctx, pathTo, checkOverrideState, checkOverrideState);
+        };
 
         this.emit('before-move', ctx, pathFrom, { pathFrom, pathTo, overwrite })
```

**Comment by AdrienCastex** [comment]
Hello,

I was unable to reproduce the error with a `VirtualFileSystem`, but I was able to produce it with a `PhysicalFileSystem`. I checked the implementation of the `PhysicalFileSystem` (the `_move` method) and I found that it was returning `true` if the resource was moved with success, otherwise `false`. But the `FileSystem` class (the `move` method) was expecting to get a `boolean` which says if the resource has been overwritten or not. Here was the mistake.

I fixed it. I still need you to check if that solved your problem (v2.3.4).

Thank you very much for your contribution.
